core/services/ocr2/plugins/ocr2keeper/integration_21_test.go

```diff
@@ -247,17 +242,21 @@ func TestIntegration_KeeperPluginLogUpkeep(t *testing.T) {
 				time.Sleep(time.Millisecond * 250) // otherwise we get "invalid transaction nonce" errors
 			}
 		})
-		// Mine enough blocks to ensre these logs don't fall into log provider range
+
+		beforeDummyBlocks := backend.Blockchain().CurrentBlock().Number.Uint64()
+
+		// Mine enough blocks to ensure these logs don't fall into log provider range
 		dummyBlocks := 500
 		for i := 0; i < dummyBlocks; i++ {
 			backend.Commit()
 			time.Sleep(time.Millisecond * 10)
 		}
-		t.Logf("Mined %d blocks, waiting for logs to be recovered", dummyBlocks)
 
-		expectedPostRecover := runs + emits
-		waitPipelineRuns(t, nodes, expectedPostRecover, testutils.WaitTimeout(t), cltest.DBPollingInterval)
+		t.Logf("Mined %d blocks, waiting for logs to be recovered", dummyBlocks)
 
+		listener, done := listenPerformedN(t, backend, registry, ids, int64(beforeDummyBlocks), len(ids))
```

Comment:
shouldn't you use `expectedPostRecover := runs + emits` instead of `len(ids)`

Reply:
`runs` is no longer available since we no longer have `checkPipelineRuns`